Source/RapyutaSimulationPlugins/Private/Drives/RobotVehicleMovementComponent.cpp

```diff
@@ -2,28 +2,25 @@
 
 #include "Drives/RobotVehicleMovementComponent.h"
 
+// UE
 #include "GameFramework/Pawn.h"
 #include "GameFramework/PlayerController.h"
 #include "Kismet/KismetMathLibrary.h"
 
-URobotVehicleMovementComponent::URobotVehicleMovementComponent()
-{
-    Gen = std::mt19937{Rng()};
-}
-
-void URobotVehicleMovementComponent::BeginPlay()
+void URobotVehicleMovementComponent::Initialize()

```

Comment:
Yeah, since now MoveComponent is created on the fly by NewObject in OnConstruction, this will be invoked there.

Reply:
Yeah, since now MoveComponent is created on the fly by NewObject in `ARobotVehicle::OnConstruction()`, this must be invoked in `ARobotVehicle::PostInitializeComponents()`.